643 - add use case template (#685)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: supporting_content/use_cases/compare_billed_cost_per_subaccount_to_budget.md

# Use Case

> Compare billed cost per subaccount to budget

Reviewer: @beau-nelford-anglepoint What changed in 1.1 that impacted this use case? None of the columns here are flagged as changed in the 1.1 changelog.

## Use Case ID

> 14383

Reviewer: Is this our number or the UC library's number? Should we reset numbers for the spec? I'm not sure how we jumped up to 14.3k use cases already 😜

> ## Supported Specification Version

Reviewer: Can we align this to the columns and move it to the end of the doc?
```suggestion
## Introduced (version)
```

Reviewer 2: Yeah having it be a introduced version makes more sense.

1.0

## Description

> Product wants to confirm charges are in line with expectations (budgets) for a business which has all its cost within a single subaccount. The ChargePeriod is used for filtering to capture items that apply to the period only (excludes adjustments from previous periods that were applied in the current BillingPeriod).

Reviewer: nit: "Product" as a pronoun is somewhat industry-specific. I would generalize to "Leadership". Also, ChargePeriod and BillingPeriod should be regular words in this context.
```suggestion
Leadership wants to confirm charges are in line with expectations (budgets) for a business which has all its cost within a single subaccount. The charge period is used for filtering to capture items that apply to the period only (excludes adjustments from previous periods that were applied in the current billing period).
```

Reviewer 2: Budgets are not an inherent part of the FOCUS spec. Refinement needed on what the actual supported feature is. For now fine, but we need to distill these down to what we actually support.

> ## 1st Degree Columns

Reviewer: nit: Should these be column names or IDs?

* Provider
* Sub Account ID
* Sub Account Name
* Billed Cost

> * Charge Category

Reviewer: Not actually needed for this use case
```suggestion
```

* Charge Period Start
* Charge Period End

## 2nd Degree Columns

## Example SQL Query
```
SELECT
  ProviderName,
  SubAccountId,
  SubAccountName,
  SUM(BilledCost) AS TotalBilledCost
FROM focus_data_table

> WHERE ChargeCategory = 'Usage'
>   AND ChargePeriodStart >= ? and ChargePeriodEnd <= ?

Reviewer: Why are we only showing usage? That doesn't seem like it's something that can be compared to a budget since purchases, adjustments, credits, and possibly taxes would impact budgets.
```suggestion
WHERE ChargePeriodStart >= ? AND ChargePeriodEnd <= ?
```

>   AND ProviderName = ?  
>   AND SubAccountId = ?  

Reviewer: nit: I'm assuming this is the exact query from the UC library, but shouldn't we leave extraneous filter conditions out? I would love to simplify this as much as possible. People should be able to figure out filtering on their own.
```suggestion
```

GROUP BY
  ProviderName,
  SubAccountId,
  SubAccountName
```

## Parameters
* ChargePeriodStart
* ChargePeriodEnd
* ProviderName
* SubAccountId

> ## Use Case Change Log
> | Version | Change | Description |
> | :-------| :------| :-----------|
> | 

Reviewer: I would avoid adding a changelog to this doc. Let's keep updates centralized in the main changelog.
```suggestion
```
